### PR TITLE
Ports: Add dtc utility suite

### PR DESCRIPTION
**Pull request by supercomputer7**
This includes the known dtc utility as well with other bundled utilities with it.

cc @ADKaster

### CODE REVIEW

File: Ports/dtc/package.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=dtc
+version=98a07006c48dc0bc3f42b3b3ce75b7f03e87e724
+workdir="dtc-${version}"
```
Comment (timschumi) [inline Ports/dtc/package.sh:4]: This is the default value, so it should be omitted.

File: Ports/dtc/package.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port=dtc
+version=98a07006c48dc0bc3f42b3b3ce75b7f03e87e724
```
Comment (timschumi) [inline Ports/dtc/package.sh:3]:
```suggestion
port='dtc'
version='98a07006c48dc0bc3f42b3b3ce75b7f03e87e724'
```

File: Ports/dtc/package.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+workdir="dtc-${version}"
+auth_type=sha256
```
Comment (timschumi) [inline Ports/dtc/package.sh:5]:
```suggestion
auth_type='sha256'
```

File: Ports/dtc/package.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+files="https://github.com/dgibson/dtc/archive/${version}.zip dtc-${version}.zip 52cffaf31085cb9dce4afb91617a7017065cff7668c9a23291ad04f8c85270c4"
```
Comment (timschumi) [inline Ports/dtc/package.sh:6]: Eh, a ZIP is technically fine, but basically everything else uses `.tar.gz`. As long as there are no files that might require their file modes to be preserved, I'll leave it up to you on whether to change it.

File: Ports/dtc/package.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+
+build() {
+    run make NO_PYTHON=1
+}
+
+install() {
+    run make PREFIX="$DESTDIR" BINDIR="$DESTDIR/usr/bin" install
```
Comment (timschumi) [inline Ports/dtc/package.sh:14]:
```suggestion
    run make PREFIX="${DESTDIR}" BINDIR="${DESTDIR}/usr/bin" install
```

**Review by timschumi — CHANGES_REQUESTED** [review]

**Review by timschumi — APPROVED** [review]
Looks good from a code-review standpoint, I'll assume that you already built and tested it anyways. :^)

**Comment by supercomputer7** [comment]
> Looks good from a code-review standpoint, I'll assume that you already built and tested it anyways. :^)

Yes I tried the dtc utility. The `fdtget` utility seems to not work because it says "Could not find required shared library: libfdt.so.1" although that library is present (so it might not be located in the correct place?). The `dtdiff` script requires `bash` so I will add that as a dependency.

**Comment by supercomputer7** [comment]
Apparently we don't support the compiled `fdtget` binary. The DynamicLoader spits out many debug messages about this:
```
97.174 fdtget(50:50): DynamicObject: DYNAMIC tag handling not implemented for DT_VERNEEDED (1879048190)
97.178 fdtget(50:50): DynamicObject: DYNAMIC tag handling not implemented for DT_VERNEEDEDNUM (1879048191)
97.182 fdtget(50:50): DynamicObject: DYNAMIC tag handling not implemented for DT_VERSYM (1879048176)
97.182 fdtget(50:50): DynamicObject: DYNAMIC tag handling not implemented for DT_VERNEEDED (1879048190)
97.182 fdtget(50:50): DynamicObject: DYNAMIC tag handling not implemented for DT_VERNEEDEDNUM (1879048191)
97.182 fdtget(50:50): DynamicObject: DYNAMIC tag handling not implemented for DT_VERSYM (1879048176)
```

**Comment by supercomputer7** [comment]
> Looks good from a code-review standpoint, I'll assume that you already built and tested it anyways. :^)

Done, the `dtdiff` script should work too now. Once CI passes this should be ready for merging :)

**Pull request merged**